### PR TITLE
Interface/81180A upgrade

### PR DESCRIPTION
**Pull request by nulinspiratie**
Changes in this PR:
- Increase maximum amplitude to 2V (which is the actual max amplitude)
- Fix bug where waveform points were not determined for each channel but only channel 1
- Add `PulseSequence.flags`, which allows flags to be added to (nested) waveforms, which can then be picked up by interfaces during the setup.
- Add feature to combine a pulse sequence into a single waveform when the flag `single_waveform` is passed. @RostyslavSavytskyy @botzec this will also be relevant to you

Requires PRs #193 and #195 to be merged first

Also not yet ready, needs some beta testing

### CODE REVIEW

File: silq/instrument_interfaces/rohde_schwarz/SGS100A_interface.py [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+
+        additional_pulses = [
+            FrequencyRampPulse(
+                name="sideband_I",
+                t_start=self.pulse.t_start,
+                t_stop=self.pulse.t_stop,
+                frequency_start=self.pulse.frequency_start - frequency,
+                frequency_stop=self.pulse.frequency_stop - frequency,
+                amplitude=amplitude,
+                phase=0,
```
Comment (RostyslavSavytskyy) [inline silq/instrument_interfaces/rohde_schwarz/SGS100A_interface.py:391]: a bit unfair to neglect the phase of the pulse, but it is depends on the application

File: silq/instrument_interfaces/rohde_schwarz/SGS100A_interface.py [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+                },
+            ),
+            FrequencyRampPulse(
+                name="sideband_Q",
+                t_start=self.pulse.t_start,
+                t_stop=self.pulse.t_stop,
+                frequency_start=self.pulse.frequency_start - frequency,
+                frequency_stop=self.pulse.frequency_stop - frequency,
+                amplitude=amplitude,
+                phase=-90,
```
Comment (RostyslavSavytskyy) [inline silq/instrument_interfaces/rohde_schwarz/SGS100A_interface.py:404]: same

File: silq/meta_instruments/layout.py [hunk 3]
```diff
@@ -1360,7 +1452,7 @@ def setup(self,
 
         # Create acquisition shapes
         trace_shapes = self.pulse_sequence.get_trace_shapes(
-            sample_rate=self.sample_rate, samples=self.samples())
+            sample_rate=self.sample_rate(), samples=self.samples())
```
Comment (RostyslavSavytskyy) [inline silq/meta_instruments/layout.py:1455]: didn't you reverse the sample_rate() calling?

**Review by RostyslavSavytskyy — APPROVED** [review]
Pull master branch first and then should be good to go

**Review by maij — COMMENTED** [review]
This is a very big PR, it looks like it's absorbed a lot of changes that are irrelevant to the 81180.
Do many of the changes present here exist in other branches (other than #193)?
At this point I'm happy to merge, pending Rosty's questions.

File: silq/meta_instruments/layout.py (hunk 3 above)
Comment (nulinspiratie) [reply silq/meta_instruments/layout.py:1455]: I believe I converted it into a parameter, so it should now have ()

File: silq/instrument_interfaces/rohde_schwarz/SGS100A_interface.py (hunk 1 above)
Comment (nulinspiratie) [reply silq/instrument_interfaces/rohde_schwarz/SGS100A_interface.py:391]: fixed

File: silq/instrument_interfaces/rohde_schwarz/SGS100A_interface.py (hunk 2 above)
Comment (nulinspiratie) [reply silq/instrument_interfaces/rohde_schwarz/SGS100A_interface.py:404]: fixed

File: silq/instrument_interfaces/keysight/E8267D_interface.py [hunk 4]
```diff
@@ -274,28 +325,28 @@ def get_additional_pulses(self, interface: InstrumentInterface):
...
+        else:  # IQ_modulation == 'on'
             if interface.FM_mode() == 'ramp':
                 assert self.pulse.frequency_sideband is not None, \
                     "Pulse.frequency_sideband must be defined when " \
                     "FM_mode = 'ramp' and IQ_modulation = 'on'"
 
                 frequency = self.pulse.frequency + self.pulse.frequency_sideband
-                frequency_difference = frequency - interface.frequency()
-                amplitude_FM = frequency_difference / interface.frequency_deviation()
+                frequency_difference = frequency - frequency
```
[the remark this answers is not in the document]
Comment (nulinspiratie) [reply silq/instrument_interfaces/keysight/E8267D_interface.py:344]: yeah definitely

**Pull request merged**